spec/polymorphic_spec.cr

```diff
@@ -0,0 +1,123 @@
+require "./spec_helper"
+
+include LazyLoadHelpers
+
+private class PolymorphicTask < BaseModel
+  table do
+    column title : String
+  end
+end
+
+private class PolymorphicTaskList < BaseModel
+  table do
+    column title : String
+  end
+end
+
+private class PolymorphicEvent < BaseModel
+  table do
+    belongs_to task : PolymorphicTask?
+    belongs_to task_list : PolymorphicTaskList?
+    polymorphic :eventable, associations: [:task, :task_list]
+  end
+end
+
+private class OptionalPolymorphicEvent < BaseModel
+  table :polymorphic_events do
+    belongs_to task : PolymorphicTask?
+    belongs_to task_list : PolymorphicTaskList?
+    polymorphic :optional_eventable, optional: true, associations: [:task, :task_list]
+  end
+end
+
+describe "polymorphic belongs to" do
+  it "sets up a method for accessing associated record" do
+    task = PolymorphicTask::SaveOperation.create!(title: "Use Lucky")
+    event = PolymorphicEvent::SaveOperation.create!(task_id: task.id)
+    event.eventable.should eq(task)
+
+    task_list = PolymorphicTaskList::SaveOperation.create!(title: "Use Lucky")
+    event = PolymorphicEvent::SaveOperation.create!(task_list_id: task_list.id)
+    event.eventable.should eq(task_list)
+  end
+
+  it "can require preloading" do
+    with_lazy_load(enabled: false) do
+      expect_raises Avram::LazyLoadError do
+        task = PolymorphicTask::SaveOperation.create!(title: "Use Lucky")
+        event = PolymorphicEvent::SaveOperation.create!(task_id: task.id)
+        event.eventable # should raise because it was not preloaded
+      end
+    end
+  end
+
+  it "has ! method to allow lazy loading" do
+    with_lazy_load(enabled: false) do
+      task = PolymorphicTask::SaveOperation.create!(title: "Use Lucky")
+      event = PolymorphicEvent::SaveOperation.create!(task_id: task.id)
+      event.eventable!.should eq(task)
+    end
+  end
+
+  it "can preload the polymorphic associations" do
+    with_lazy_load(enabled: false) do
+      task = PolymorphicTask::SaveOperation.create!(title: "Use Lucky")
+      event = PolymorphicEvent::SaveOperation.create!(task_id: task.id)
```

Comment:
Nit: probably don't need to save this variable.

Reply:
It is used in the next line. But maybe I misunderstood?

Reply:
Oh sorry, missed that. I think it threw me that it was defined twice. Nevermind.